XDMA/linux-kernel/xdma/tsn.c

```diff
@@ -379,22 +376,27 @@ static bool get_timestamps(struct timestamps* timestamps, const struct tsn_confi
 	return true;
 }
 
-static bool append_buffer_track(struct buffer_tracker* tracker, sysclock_t free_at) {
-	if (tracker->count >= HW_QUEUE_SIZE) {
+static bool append_buffer_track(struct buffer_tracker* buffer_tracker) {
+	if (buffer_tracker->entry_count >= HW_QUEUE_SIZE) {
 		return false;
 	}
-	tracker->free_at[tracker->head] = free_at;
-	tracker->head = (tracker->head + 1) % HW_QUEUE_SIZE;
-	tracker->count += 1;
+
+	buffer_tracker->entry_count += 1;
 	return true;
 }
 
-static bool cleanup_buffer_track(struct buffer_tracker* tracker, sysclock_t now) {
-	while (tracker->count > 0 && tracker->free_at[tracker->tail] < now) {
-		tracker->tail = (tracker->tail + 1) % HW_QUEUE_SIZE;
-		tracker->count -= 1;
+void tsn_pop_buffer_track(struct pci_dev* pdev) {
```

Comment:
Changing its name to _pop_ is something not suitable. While _cleanup_ is still fine, As this function reads hw reg and update current state, _update_ also good